plugins/out_pgsql/pgsql_connections.c

```diff
@@ -0,0 +1,184 @@
+/* -*- Mode: C; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- */
+
+/*  Fluent Bit
+ *  ==========
+ *  Copyright (C) 2020 The Fluent Bit Authors
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+#include "pgsql.h"
+
+void pgsql_destroy_connections(struct flb_pgsql_config *ctx)
+{
+    struct mk_list *tmp;
+    struct mk_list *head;
+    struct flb_pgsql_conn *conn;
+    PGresult *res = NULL;
+
+    mk_list_foreach_safe(head, tmp, &ctx->conn_queue) {
+        conn = mk_list_entry(head, struct flb_pgsql_conn, _head);
+        if(PQstatus(conn->conn) == CONNECTION_OK) {
+            while(PQconsumeInput(conn->conn) == 0) {
+                res = PQgetResult(conn->conn);
+                if(PQresultStatus(res) != PGRES_COMMAND_OK) {
+                    flb_plg_warn(ctx->ins, "%s",
+                                 PQerrorMessage(conn->conn));
+                }
+                PQclear(res);
+            }
+        }
+        PQfinish(conn->conn);
+        flb_free(conn);
+    }
+}
+
+void *pgsql_create_connection(struct flb_pgsql_config *ctx)
+{
+    struct flb_pgsql_conn *conn;
+
+    conn = flb_calloc(1, sizeof(struct flb_pgsql_conn));
+    conn->conn = PQsetdbLogin(ctx->db_hostname,
+                              ctx->db_port,
+                              NULL, NULL,
+                              ctx->db_name,
+                              ctx->db_user,
+                              ctx->db_passwd);
+
+    if(PQstatus(conn->conn) != CONNECTION_OK) {
+        flb_plg_error(ctx->ins,
+                      "failed connecting to host=%s with error: %s",
+                      ctx->db_hostname, PQerrorMessage(conn->conn));
+        PQfinish(conn->conn);
+        flb_free(conn);
+        return NULL;
+    }
+
+    flb_plg_info(ctx->ins, "switching postgresql connection "
+                 "to non-blocking mode");
+
+    if(PQsetnonblocking(conn->conn, 1) != 0) {
+        flb_plg_error(ctx->ins, "non-blocking mode not set");
+        PQfinish(conn->conn);
+        flb_free(conn);
+        return NULL;
+    }
+
+    return conn;
+}
+
+int pgsql_start_connections(struct flb_pgsql_config *ctx)
+{
+    int i;
+    struct flb_pgsql_conn *conn = NULL;
+
+    mk_list_init(&ctx->conn_queue);
+    ctx->active_conn = 0;
+
+    for(i = 1; i <= ctx->min_pool_size; i++) {
```

Comment:
please adjust to ```i = 0; i < ctx->min_pool_size```, it's the same thing but just trying to follow how iterators are implemented in other components of the project